app/controllers/concerns/hyrax/collections_controller_behavior.rb

```diff
@@ -25,7 +25,8 @@ module CollectionsControllerBehavior
     end
 
     def show
-      @curation_concern ||= Hyrax.query_service.find_by_alternate_identifier(alternate_identifier: params[:id], use_valkyrie: false)
+      valkyrie_obj = Hyrax.query_service.find_by_alternate_identifier(alternate_identifier: params[:id])
+      @curation_concern ||= Wings::ActiveFedoraConverter.new(resource: valkyrie_obj).convert
```

Comment:
This is a good start.

Have you looked at how possible it might be to allow `presenter` and/or `query_collection_members` to work using the Valkyrie::Resource? It would be good to push the `#convert` call as far from the retrieve as possible, or at least to identify the specific blockers to doing so.

Reply:
Very good question!
I looked into #present method and #query_collection_members, and could confirm that it is not relevant to this issue.
The only issue exist in @curation_concern
And i also checked the process where the @curation_concern got parsed to response body is out of the scope of hyrax codebase, which means we can not modify it.

Please see the debugging infor and let me know your thoughts: https://github.com/samvera/hyrax/issues/3865#issuecomment-517474571

Reply:
> I looked into #present method and #query_collection_members, and could confirm that it is not relevant to this issue.

i think it's really core to the issue to identify the changes that would be needed to these methods to make them work with valkyrie models. i understand that to be the main point of the work.

> And i also checked the process where the @curation_concern got parsed to response body is out of the scope of hyrax codebase, which means we can not modify it.

we're going to need to modify it sooner or later to complete the Wings work. can we at least document where we reach past the Hyrax codebase and open an issue for resolving it?

Reply:
I will double check it.

Reply:
to be clear: the ultimate goal is to avoid needing the `Wings::ActiveFedoraConverter.new(resource: valkyrie_obj).convert` call anywhere, electing to use the `Valkyrie::Resource` result everywhere instead. I expect that there will be places where this won't be practical yet, but the ticket requires pushing as far as is reasonable, and documenting the current boundaries as upcoming work.

Reply:
There is no difference between use_valkyrie true or false for `presenter` object.

But you are right, I found this one is relevant: 
https://github.com/samvera/hyrax/blob/master/app/views/hyrax/base/show.json.jbuilder

```
(byebug) json.extract! @curation_concern, *[:id] + @curation_concern.class.fields.reject { |f| [:has_model].include? f }
[:id, :id, :internal_resource, :created_at, :updated_at, :read_groups, :read_users, :edit_users, :edit_groups, :alternate_ids, :embargo, :lease, :member_of_collection_ids, :member_ids, :create_date, :modified_date, :depositor, :title, :date_uploaded, :date_modified, :head, :tail, :collection_type_gid, :alt_title, :label, :relative_path, :import_url, :resource_type, :creator, :contributor, :description, :abstract, :keyword, :license, :rights_notes, :rights_statement, :access_right, :publisher, :date_created, :subject, :language, :identifier, :based_near, :related_url, :bibliographic_citation, :source, :access_control_ids, :list_source_ids, :related_object_ids, :representative_ids, :thumbnail_ids, :access_control_id, :representative_id, :thumbnail_id]
```
But i'm not sure why it returns only key instead of key value. I will dig into it next sprint.

Reply:
> There is no difference between use_valkyrie true or false for presenter object.

this is fantastic news!